src/server/cards/adventures/Hero.js

```diff
@@ -0,0 +1,35 @@
+import Card from 'cards/Card';
+
+export default class Hero extends Card {
+  static cost = new Card.Cost({ coin: 5 });
+  static types = new Set(['Action', 'Traveller']);
+  async onPlay(player) {
+    player.money += 2;
+    const [supply] = await player.selectSupplies({
+      min: 1,
+      max: 1,
+      predicate: s => (
+        s.cards.size > 0 &&
+        s.cards.last().types.has('Treasure')
+      ),
+      message: 'Choose a Treasure card to gain'
+    });
+    await player.gain(supply.title);
```

Comment:
if (supply), I mean there should almost always be a supply but this sanity check is important.

What if we have a five player game and the Copper + Silver + Gold piles run out and some one plays Hero? Then what? Happens more often than you would think.